src/data/java/com/github/minecraftschurlimods/arsmagicalegacy/data/AMRitualProvider.java

```diff
@@ -83,63 +82,52 @@ protected void addRituals(BiConsumer<ResourceLocation, Ritual> consumer) {
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_air_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.AIR_GUARDIAN_SPAWN_RITUAL))
                 .with(ItemDropRitualTrigger.item(AMItems.TARMA_ROOT.get()))
                 .with(new HeightRequirement(MinMaxBounds.Ints.atLeast(128)))
                 .with(new EntitySpawnRitualEffect(AMEntities.AIR_GUARDIAN.get()))
                 .build(consumer);
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_earth_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.EARTH_GUARDIAN_SPAWN_RITUAL))
-                .with(ItemDropRitualTrigger.tags(
-                        Tags.Items.GEMS_EMERALD,
-                        AMTags.Items.GEMS_CHIMERITE,
-                        AMTags.Items.GEMS_TOPAZ))
+                .with(ItemDropRitualTrigger.tags(Tags.Items.GEMS_EMERALD, AMTags.Items.GEMS_CHIMERITE, AMTags.Items.GEMS_TOPAZ))
                 .with(new EntitySpawnRitualEffect(AMEntities.EARTH_GUARDIAN.get()))
                 .build(consumer);
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_fire_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.FIRE_GUARDIAN_SPAWN_RITUAL))
                 .with(ItemDropRitualTrigger.stackExact(ArsMagicaAPI.get().getAffinityHelper().getEssenceForAffinity(IAffinity.WATER)))
                 .with(new UltrawarmDimensionRequirement())
                 .with(new EntitySpawnRitualEffect(AMEntities.FIRE_GUARDIAN.get()))
                 .build(consumer);
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_water_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.WATER_GUARDIAN_SPAWN_RITUAL))
                 .with(ItemDropRitualTrigger.ingredients(Ingredient.of(ItemTags.BOATS), Ingredient.of(Items.WATER_BUCKET)))
                 .with(new BiomeRequirement(AMTags.Biomes.CAN_SPAWN_WATER_GUARDIAN))
                 .with(new EntitySpawnRitualEffect(AMEntities.WATER_GUARDIAN.get()))
                 .build(consumer);
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_ender_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.ENDER_GUARDIAN_SPAWN_RITUAL))
                 .with(ItemDropRitualTrigger.item(Items.ENDER_EYE))
                 .with(new EnderDragonDimensionRequirement())
                 .with(new EntitySpawnRitualEffect(AMEntities.ENDER_GUARDIAN.get()))
                 .build(consumer);
-
         builder(new ResourceLocation(ArsMagicaAPI.MOD_ID, "spawn_ice_guardian"))
                 .with(new RitualStructureRequirement(PatchouliCompat.ICE_GUARDIAN_SPAWN_RITUAL))
                 .with(new EntitySummonTrigger(EntityType.SNOW_GOLEM))
-                /*.with(new BiomeRequirement(Tags.Biomes.IS_FROZEN))*///TODO: Add biome requirement
+//                .with(new BiomeRequirement(Tags.Biomes.IS_FROZEN))
```

Comment:
why remove the todo

Reply:
I changed the comment to single-line and forgot to readd the todo